packages/typespec-python/test/azure/generated/azure-mgmt-spheredpg/azure/mgmt/spheredpg/_model_base.py

```diff
@@ -560,6 +572,27 @@ def _as_dict_value(v: typing.Any, exclude_readonly: bool = False) -> typing.Any:
             return {dk: Model._as_dict_value(dv, exclude_readonly=exclude_readonly) for dk, dv in v.items()}
         return v.as_dict(exclude_readonly=exclude_readonly) if hasattr(v, "as_dict") else v
 
+    def __getattr__(self, name: str) -> typing.Any:
+        if not name.startswith("__"):
+            if "properties" in self._attr_to_rest_field:
+                if self.properties:
```

Comment:
I think all we need here is 

```python
if "properties" in self._attr_to_rest_field:
  if self.properties:
    return getattr(self.properties, name)
  return None
```

Because we should only enter into this line of code if `self.properties` exists and is `None`. Thoughts?

Reply:
Only when `name` is part of `properties` we return value or None. Otherwise, we need to raise AttributeError

https://github.com/Azure/autorest.python/blob/fe80efe0ec224b179e8d110a87de6d21d2daf61b/packages/typespec-python/test/azure/mock_api_tests/test_mgmt_models_mode.py#L64-L66